src/dfx/src/commands/quickstart.rs

```diff
@@ -0,0 +1,130 @@
+use anyhow::{bail, Context, Error};
+use candid::Principal;
+use dialoguer::{Confirm, Input};
+use ic_agent::Identity as _;
+use ic_utils::interfaces::{
+    management_canister::builders::InstallMode, ManagementCanister, WalletCanister,
+};
+use indicatif::ProgressBar;
+use num_traits::Inv;
+use rust_decimal::Decimal;
+use tokio::runtime::Runtime;
+
+use crate::{
+    commands::ledger::{create_canister::MEMO_CREATE_CANISTER, notify_create, transfer_cmc},
+    lib::{
+        environment::Environment,
+        error::DfxResult,
+        identity::{Identity, IdentityManager},
+        ledger_types::{Memo, NotifyError},
+        nns_types::{
+            account_identifier::AccountIdentifier,
+            icpts::{ICPTs, TRANSACTION_FEE},
+        },
+        operations::{
+            canister::install_wallet,
+            ledger::{balance, xdr_permyriad_per_icp},
+        },
+        provider::create_agent_environment,
+        waiter::waiter_with_timeout,
+    },
+    util::{assets::wallet_wasm, expiry_duration},
+};
+
+pub fn exec(env: &dyn Environment) -> DfxResult {
+    let env = create_agent_environment(env, Some("ic".to_string()))?;
+    let agent = env.get_agent().expect("Unable to create agent");
+    let ident = IdentityManager::new(&env)?.instantiate_selected_identity()?;
```

Comment:
When running this with a non-default identity, the command prompts the user two times to enter their decryption password.

The AgentEnvironment implements `get_selected_identity` and `get_selected_identity_principal` for use in cases like this.

```
$ cargo run quickstart --identity qstest
Please enter a passphrase for your identity: [hidden]
Decryption complete.
Please enter a passphrase for your identity: [hidden]
Decryption complete.
Your DFX user principal: r5eou-p7obc-fbah7-nnkyo-yncd3-hmsd5-qipan-2h5va-ittct-j2lcs-vae

```